src/Panel/Ui/Stat.php

```diff
@@ -0,0 +1,133 @@
+<?php
+
+namespace Kirby\Panel\Ui;
+
+use Kirby\Cms\ModelWithContent;
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Toolkit\I18n;
+
+/**
+ * @package   Kirby Panel
+ * @author    Bastian Allgeier <bastian@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://getkirby.com/license
+ * @since     5.1.0
+ */
+class Stat extends Component
+{
+	public function __construct(
+		public ModelWithContent $model,
+		public array|string $label,
+		public array|string $value,
+		public string $component = 'k-stat',
+		public string|null $icon = null,
+		public array|string|null $info = null,
+		public array|string|null $link = null,
+		public string|null $theme = null,
+	) {
+	}
+
+	public static function from(
+		ModelWithContent $model,
+		array|string $input
+	): static {
+		if (is_string($input) === true) {
+			return static::fromQuery(
+				model: $model,
+				query: $input
+			);
+		}
+
+		return new static(...[
+			...$input,
+			'model' => $model,
+			'label' => $input['label'],
+			'value' => $input['value'],
+		]);
```

Comment:
`Suggestion:` Have you already tried if this works?

```suggestion
		$input['model'] = $model;
		return new static(...$input);
```

Reply:
I just tried and Psalm isn't happy about it. It feels a bit weird indeed to list them here and potentially have notice errors. Maybe defaulting to null could be another option and then the class constructor will throw a usable error?

Reply:
If it's just psalm, we might also want to suppress the psalm error, if we actually prefer our syntax there.

Reply:
you are right. It's weird to make the machine happy and us sad 😆